lib/workers/evm_server.rb

```diff
@@ -38,7 +38,8 @@ def start
   end
 
   def start_servers
-    refresh_servers_to_monitor
+    @servers_to_monitor = servers_from_db if servers_to_monitor.empty?
```

Comment:
I like this change better than the prior one and the suggested one I came up with [here](https://github.com/ManageIQ/manageiq/pull/21339#discussion_r673473818).  This change is a mixture of Jason's suggestion to move the initialization of this list to the `start_servers` method plus me reviewing that code and realizing that `refresh_servers_to_monitor` should no longer be needed once the above is added as the former becomes a no-op because `servers_from_db == servers_to_monitor`:

```ruby
  def refresh_servers_to_monitor
    servers_to_start    = servers_from_db    - servers_to_monitor # should be empty
    servers_to_shutdown = servers_to_monitor - servers_from_db # should be empty

    servers_to_start.each do |s| # nothing to do here
      servers_to_monitor << s
      impersonate_server(s)
      start_server
    end

    servers_to_shutdown.each do |s| # nothing to do here
      servers_to_monitor.delete(s)
      s.shutdown
    end
  end
```